common/src/main/java/org/opensearch/ml/common/utils/StringUtils.java

```diff
@@ -48,13 +45,16 @@ public static String toUTF8(String rawString) {
         return utf8EncodedString;
     }
 
-    public static Map<String, Object> fromJson(String jsonStr, String defaultKey) {
+    public static Map<String, Object> fromJson(String input, String defaultKey) {
+        if (!isJson(input)) {
+            return Collections.singletonMap(defaultKey, input);
```

Comment:
By adding this if branch, this function extends to support more cases which is not Map or List. That's changing the original design. Will it cause any issue? Why we have to do this?

Reply:
The original design is only accepting json format response? If so, it's ok to remove this.